servatrice/src/serversocketinterface.cpp

```diff
@@ -747,6 +749,45 @@ Response::ResponseCode ServerSocketInterface::cmdReplayDeleteMatch(const Command
 // MODERATOR FUNCTIONS.
 // May be called by admins and moderators. Permission is checked by the calling function.
 
+Response::ResponseCode ServerSocketInterface::cmdGetBanHistory(const Command_GetBanHistory &cmd, ResponseContainer &rc)
+{
+    if (!sqlInterface->checkSql())
+        return Response::RespInternalError;
+
+    QString userName = QString::fromStdString(cmd.user_name());
+
+    //lookup users ban history from database
+    QSqlQuery *query = sqlInterface->prepareQuery("SELECT A.id_admin, A.time_from, A.minutes, A.reason, A.visible_reason, B.name AS name_admin FROM {prefix}_bans A LEFT JOIN {prefix}_users B ON A.id_admin=B.id WHERE A.user_name = :user_name");
```

Comment:
Rather than a comment it would be more flexible to have a `getUserBanHistory` method. It's self-explanatory at that point and it provides us an easy method to work with when we get to automated testing.